tests/test_schedules.py

```diff
@@ -401,37 +401,37 @@ def foo():
         fission(foo, foo.find("x = 0.0").after(), n_lifts=2)
 
 
-def test_shrink_dim(golden):
+def test_resize_dim(golden):
     @proc
     def foo():
         x: i8[10]
         for i in seq(1, 9):
             x[i] = 1.0
 
-    foo = shrink_dim(foo, "x", 0, 1, 9)
+    foo = resize_dim(foo, "x", 0, 19, 1)
     assert str(simplify(foo)) == golden
 
     with pytest.raises(SchedulingError, match="The buffer x is accessed out-of-bounds"):
-        foo = shrink_dim(foo, "x", 0, 1, 8)
+        foo = resize_dim(foo, "x", 0, 7, 1)
     with pytest.raises(SchedulingError, match="The buffer x is accessed out-of-bounds"):
-        foo = shrink_dim(foo, "x", 0, 2, 9)
+        foo = resize_dim(foo, "x", 0, 7, 2)
 
 
-def test_shrink_dim_2(golden):
+def test_resize_dim_2(golden):
     @proc
     def foo(n: size):
         assert n > 4
         x: i8[n]
         for i in seq(2, n - 1):
             x[i] = 1.0
 
-    foo = shrink_dim(foo, "x", 0, 2, "n-1")
+    foo = resize_dim(foo, "x", 0, "n-3", 2)
     assert str(simplify(foo)) == golden
 
     with pytest.raises(SchedulingError, match="The buffer x is accessed out-of-bounds"):
-        foo = shrink_dim(foo, "x", 0, 2, "n-2")
+        foo = resize_dim(foo, "x", 0, "n-4", 2)
     with pytest.raises(SchedulingError, match="The buffer x is accessed out-of-bounds"):
-        foo = shrink_dim(foo, "x", 0, 3, "n-1")
+        foo = resize_dim(foo, "x", 0, "n-4", 3)
```

Comment:
Can you add a test case where `offset` is not an integer constant? (e.g., `i-5` or sth?)

Reply:
I think that implicitly exists in Halide test cases, but yes I can add one here too.